inject/src/main/java/io/micronaut/context/DefaultBeanContext.java

```diff
@@ -3158,13 +3155,40 @@ public boolean test(BeanDefinition<T> candidate) {
             if (candidates.size() == 1) {
                 return candidates.iterator().next();
             } else {
-                Collection<BeanDefinition<T>> exactMatches = filterExactMatch(beanClass, candidates);
-                if (exactMatches.size() == 1) {
-                    definition = exactMatches.iterator().next();
-                } else if (throwNonUnique) {
-                    definition = findConcreteCandidate(beanClass, specifiedQualifier, candidates);
+                if (candidates.stream().anyMatch(candidate -> candidate.hasAnnotation(Order.class))) {
+                    // pick the bean with the highest priority
+                    final Iterator<BeanDefinition<T>> i = candidates.stream()
+                            .sorted((bean1, bean2) -> {
+                                int order1 = OrderUtil.getOrder(bean1, bean1);
+                                int order2 = OrderUtil.getOrder(bean2, bean1);
```

Comment:
Is this a typo?  `bean2, bean1`

Should this be:  `int order2 = OrderUtil.getOrder(bean2, bean2);`